index.d.ts

```diff
@@ -130,14 +135,15 @@ export class VM {
  * to any VM (context); rather, it is bound before each run, just for that run.
  */
 export class VMScript {
-  constructor(code: string, path?: string, options?: {
-    lineOffset: number;
-    columnOffset: number;
-  });
-  /** Wraps the code */
-  wrap(prefix: string, postfix: string): VMScript;
-  /** Compiles the code. If called multiple times, the code is only compiled once. */
-  compile(): any;
+  code: string;
+  filename: string;
+  options: VMScriptOptions;
+  /** Create VMScript instance. */
+  constructor(code: string, filename?: string, options?: VMScriptOptions);
+  /** Wraps the code. Will invalidate the code cache. */
+  wrap(prefix: string, suffix: string): VMScript;
+  /** This code will be compiled to VM code. */
+  compile(): VMScript;
```

Comment:
Can't typescript use `this` as type if it returns `this`? Wouldn't that be even more specific?

Reply:
It can, but I don't think it makes a difference in this case.
https://www.typescriptlang.org/docs/handbook/advanced-types.html#polymorphic-this-types

Reply:
If someone extends VMScript it would do.